system/ee/ExpressionEngine/Library/CP/URL.php

```diff
@@ -225,7 +241,7 @@ public function __unserialize($serialized)
      */
     public function unserialize($serialized)
     {
-        $data = json_decode($serialized, true);
+        $data = unserialize($serialized);
 
         if (! is_array($data)) {
```

Comment:
@intoeetive similarly I think we could dry this up by just doing `return $this->__unserialize($data);` after line 244